### ISSUE TITLE
Facing session not created error

### ISSUE THREAD

**Issue by darshan-devloper** [espressif/esp-idf-provisioning-android#101]
Hi, 

We are using the 2.1.2 version of the library in Android. When we try to fetch the SSID list or connect with WIFI using manually enter SSID and password we get a **_Failed to create session error_** in the scanNetworks Method. Can you please help us with that? We are using Security 1. 

We are trying to provision the device in BLE mode.

**Comment by KhushbuShah25** [comment]
Hi @darshan-devloper , I have tested with Security 1 and 2. I am able to create session and getting Wi-Fi list.
Can you give some details about device firmware ? Which IDF example you are using ?

**Comment by darshan-devloper** [comment]
Hi, @KhushbuShah25  I tried with Local SDK itself. It's working fine in Debug mode but in Release mode it gives an error at **getStep0Request Method of Security1.java** in the below line of code

  Sec1.SessionCmd0 sessionCmd0 = Sec1.SessionCmd0
                        .newBuilder()
                        .setClientPubkey(ByteString.copyFrom(this.publicKey))
                        .build();
                        
 The error Is as below

**java.lang.RuntimeException: Field clientPubkey_ for gj.t0 not found. Known fields are [private hf.i gj.t0.l, private static final gj.t0 gj.t0.m, private static volatile hf.a1 gj.t0.n]**

I tried to run the same code in Debug and Release mode. Debug mode works perfectly but Release mode gives an error. 

Can we connect so I can give you a complete idea of the code?

**Comment by KhushbuShah25** [comment]
Ok. I will check with release mode and let you know.

**Comment by CrownedComedian** [comment]
Hi @darshan-devloper any update on this issue?  I am facing the same issue on the Android release build version of my app.  Even after upgrading to v2.2.1 of the project dependency.

**Comment by KhushbuShah25** [comment]
Hi @CrownedComedian , there may be minify flag issue. 
Can you check if [minify issue](https://github.com/espressif/esp-idf-provisioning-android/issues/68) is the same reason for this issue or not ?

**Comment by CrownedComedian** [comment]
Hi @KhushbuShah25 I was just looking into this from another issue when I saw your comment about it.  Yes, adding those flags fixed the issue for me, thanks for the quick reply!!!

For anyone else, I just added the `minifyEnabled `and `shrinkResources `lines to my project level build.gradle file.  It now looks something like this:
```
android {
    ...
    buildTypes {
        release {
            minifyEnabled false
            shrinkResources false
            signingConfig signingConfigs.release
        }
    }
}
```

**Comment by gintsgints** [comment]
Will this problem stay and we have to leave minimifyEnabled = false ? Or it is planed to fix it somehow else?

**Comment by KhushbuShah25** [comment]
minifyEnabled issue is already fixed in commit `342ce0abc280e794bc281a2a83ec9cd612dc294a`.
Are you still facing issue with latest lib ?

**Comment by gintsgints** [comment]
Yes. With newest version this is fixed! Thank you. IMHO issue can be closed.
